satpy/readers/seviri_l1b_nc.py

```diff
@@ -60,29 +61,46 @@ def end_time(self):
         """Get the end time."""
         return self.deltaEnd
 
-    def _read_file(self):
+    @cached_property
+    def nc(self):
         """Read the file."""
-        if self.nc is None:
+        f_obj = open_file_or_filename(self.filename)
 
-            self.nc = xr.open_dataset(self.filename,
-                                      decode_cf=True,
-                                      mask_and_scale=False,
-                                      chunks=CHUNK_SIZE)
+        dataset = xr.open_dataset(f_obj,
+                                  decode_cf=True,
+                                  mask_and_scale=False,
+                                  chunks=CHUNK_SIZE)
 
+        return dataset.rename({'num_columns_vis_ir': 'x', 'num_rows_vis_ir': 'y'})
+
+    def get_metadata(self):
+        """Get metadata."""
         # Obtain some area definition attributes
-        equatorial_radius = (self.nc.attrs['equatorial_radius'] * 1000.)
+        equatorial_radius = self.nc.attrs['equatorial_radius'] * 1000.
         polar_radius = (self.nc.attrs['north_polar_radius'] * 1000 + self.nc.attrs['south_polar_radius'] * 1000) * 0.5
         ssp_lon = self.nc.attrs['longitude_of_SSP']
+        self.mda['vis_ir_grid_origin'] = self.nc.attrs['vis_ir_grid_origin']
+        self.mda['vis_ir_column_dir_grid_step'] = self.nc.attrs['vis_ir_column_dir_grid_step'] * 1000.0
+        self.mda['vis_ir_line_dir_grid_step'] = self.nc.attrs['vis_ir_line_dir_grid_step'] * 1000.0
+        # if FSFile is used h5netcdf engine is used which outputs arrays instead of floats for attributes
+        if isinstance(equatorial_radius, np.ndarray):
+            equatorial_radius = equatorial_radius[0]
+            polar_radius = polar_radius[0]
+            ssp_lon = ssp_lon[0]
+            self.mda['vis_ir_column_dir_grid_step'] = self.mda['vis_ir_column_dir_grid_step'][0]
+            self.mda['vis_ir_line_dir_grid_step'] = self.mda['vis_ir_line_dir_grid_step'][0]
+
         self.mda['projection_parameters'] = {'a': equatorial_radius,
                                              'b': polar_radius,
                                              'h': 35785831.00,
                                              'ssp_longitude': ssp_lon}
 
-        self.mda['number_of_lines'] = int(self.nc.dims['num_rows_vis_ir'])
-        self.mda['number_of_columns'] = int(self.nc.dims['num_columns_vis_ir'])
+        self.mda['number_of_lines'] = int(self.nc.dims['y'])
+        self.mda['number_of_columns'] = int(self.nc.dims['x'])
 
-        self.mda['hrv_number_of_lines'] = int(self.nc.dims['num_rows_hrv'])
-        self.mda['hrv_number_of_columns'] = int(self.nc.dims['num_columns_hrv'])
+        # Only needed for HRV channel which is not implemented yet
+        # self.mda['hrv_number_of_lines'] = int(self.nc.dims['num_rows_hrv'])
+        # self.mda['hrv_number_of_columns'] = int(self.nc.dims['num_columns_hrv'])
```

Comment:
Will you implement this in the current PR ?

Reply:
Well I have been thinking about implementing the HRV reader for a while but never came around to it since I rarely use it. I was not planning to implement it with this PR. I just refactored a little bit because there was some code regarding HRV including these commented out lines which was not actually needed since it is not implemented. I wanted to keep these lines for reference to the dim names. But the could be removed currently.

Reply:
They can stay, that's fine. And the HRV should be another PR anyway, I realise now it's quite some work.